src/main/java/hudson/plugins/analysis/core/PluginDescriptor.java

```diff
@@ -122,21 +128,6 @@ public PluginDescriptor(final Class<? extends Publisher> clazz) {
     }
 
     @Override
-    public Publisher newInstance(final StaplerRequest req, final JSONObject formData)
-            throws hudson.model.Descriptor.FormException {
-        JSONObject converted;
-        if (formData.containsKey(CONFIGURATION_SECTION_KEY)) {
-            JSONObject old = formData.getJSONObject(CONFIGURATION_SECTION_KEY);
-            formData.put(CONFIGURATION_SECTION_KEY, convertHierarchicalFormData(old));
-            converted = formData;
-        }
-        else {
-            converted = convertHierarchicalFormData(formData);
-        }
-        return super.newInstance(req, converted);
-    }
```

Comment:
OK, this works given the existing field structure. The cleaner design is to use `f:optionalProperty` with a nested `Describable` that holds all the fields related to the _Compute new warnings (based on reference build)_ block. Then either the field with this nested object is null, when the mode is disabled, or it is non-null and its value specifies the behavior. Might be a desirable refactoring, but out of scope here.